front/pages/api/stripe/webhook.ts

```diff
@@ -277,10 +283,66 @@ async function handler(
             { event },
             "[Stripe Webhook] Received customer.subscription.updated event."
           );
+          stripeSubscription = event.data.object as Stripe.Subscription;
+          const previousAttributes = event.data.previous_attributes;
+          if (!previousAttributes) break; // should not happen by definition of the subscription.updated event
+
+          if (
+            // The subscription is canceled (but not yet ended) or reactivated
+            stripeSubscription.status === "active" &&
+            "cancel_at_period_end" in previousAttributes
+          ) {
+            // first update subscription endDate in our database
+            // cancel_at set means the user just canceled, unset means the user just reactivated
+            const endDate = stripeSubscription.cancel_at
+              ? new Date(stripeSubscription.cancel_at * 1000)
+              : null;
+
+            // get subscription
+            const subscription = await Subscription.findOne({
+              where: { stripeSubscriptionId: stripeSubscription.id },
+              include: [Workspace],
```

Comment:
I didn't know about this short syntax 🤩

Reply:
neither did I ! thanks copilot :smile: (i checked and tested ofc)